nuxeo-drive-client/nxdrive/osi/windows/windows.py

```diff
@@ -143,19 +185,49 @@ def register_protocol_handlers(self):
             [('', _winreg.REG_SZ, command)],
         )
 
-    def _recursive_delete(self, reg, start_path, end_path):
-        try:
-            while (len(start_path) < len(end_path)):
-                _winreg.DeleteKey(reg, end_path)
-                end_path = end_path[0:end_path.rfind('\\')]
-        except Exception, e:
-            pass
+    def traverse_path(self, reg, start_path, access_mask, enrties_list):
+        key = _winreg.OpenKey(reg, start_path, 0, access_mask)
+        count = 0
+        child_count = _winreg.QueryInfoKey(key)[0]
+        while count < child_count:
+            reg_name = _winreg.EnumKey(key, count)
+            self.traverse_path(reg, start_path + '\\' + reg_name, access_mask, enrties_list)
+            enrties_list.append(start_path + '\\' + reg_name)
+            count = count + 1
+        _winreg.CloseKey(key)
+
+    '''
+        delete registry entries added for windows 10 navigation panel
+    '''
+
+    def _delete_registry_entries(self):
+        reg = _winreg.ConnectRegistry(None, _winreg.HKEY_CURRENT_USER)
+        path = 'Software\\Classes\\CLSID'
+        self._recursive_delete(reg, path, self.guid)
+        self._recursive_delete(reg, self.EXPLORER + '\Desktop\NameSpace', self.guid)
+        self._delete_reg_value(reg, self.EXPLORER + '\HideDesktopIcons\NewStartPanel', self.guid)
+
+    def _recursive_delete(self, reg, start_path, name):
```

Comment:
Accept Path from which registry keys needs to be delete, traverse till the depth.
First delete the keys in depth and then the ancestor.